### PR TITLE
Feature/trees branch

### PR DESCRIPTION
**Pull request by solidajenjo**

### CODE REVIEW

File: Game/config.json [hunk 1]
```diff
@@ -1 +1,2 @@
...
\ No newline at end of file
+{"window":{"fullscreen":0,"resizable":1,"borderless":0,"fullscreen_desktop":0,"width":1604,"height":881,"brightness":1.0},"renderer":{"msaa":1,"msaa_level":2,"picker_debug":0,"light_debug":0,"quadtree_debug":0,"grid_debug":0,"current_scale":100,"depthTest":1,"wireframe":0,"vsync":0,"frustumMainCamera":0,"skybox":1},"camera":{"position":[5463.80810546875,2945.64599609375,1899.845947265625],"forward":[-0.7880058288574219,-0.44004711508750918,-0.43058741092681887],"up":[-0.38615167140960696,0.8979781270027161,-0.21100297570228578],"movement":15.0,"rotation":200.0,"zoom":0.10000000149011612,"fov":0.6981316804885864,"znear":10.0,"zfar":100000.0},"textures":{"filter":3},"scene":{"sphereUID":4069615675,"cubeUID":3441925136,"ambient":[0.24590164422988892,0.24589918553829194,0.24589918553829194],"defaultscene":"UnityScene"}}
```
Comment (Rubetoman) [inline Game/config.json]: Avoid uploading changes to config.ini 😃

File: Game/imgui.ini [hunk 2]
```diff
@@ -21,8 +21,9 @@ Collapsed=0
 [Window][Scene]
 Pos=380,23
 Size=941,706
+
 Collapsed=0
-DockId=0x00000001,0
```
Comment (Rubetoman) [inline Game/imgui.ini]: Avoid uploading changes to imgui.ini 😃

Comment (solidajenjo) [reply Game/imgui.ini]: Reverted changes

File: Game/config.json (hunk 1 above)
Comment (solidajenjo) [reply Game/config.json]: Reverted changes

File: Source/AABBTree.cpp [hunk 3]
```diff
@@ -1,38 +1,231 @@
...
+		if (node->rightSon != nullptr)
+			S.push(node->rightSon);
 
-	if (rootIndex == -1 || nodes[rootIndex] == nullptr)
+		dd::aabb(node->aabb.minPoint, node->aabb.maxPoint, color);
+
+	}
+	*/
+}
+inline void AABBTree::RecalculateBoxes(AABBTreeNode* node)
```
Comment (AlbertVVila) [inline Source/AABBTree.cpp]: Don't make methods with loops inside inline

File: Source/AABBTree.h [hunk 4]
```diff
@@ -1,41 +1,99 @@
...
+	AABBTreeNode* treeRoot = nullptr;
+	AABBTreeNode** nodesFreePool = nullptr;
+	AABBTreeNode** nodesCreatedPool = nullptr; //all created nodes container to cleaning pourposes
+	bool showOnEditor = false;
+	unsigned lastFreeNode = MAX_AABB_TREE_NODES - 1;
 };
 
-class AABBTree
+template<typename T>
+inline void AABBTree::GetIntersections(T &intersector, std::unordered_set<GameObject*> &intersections)
```
Comment (AlbertVVila) [inline Source/AABBTree.h]: Too large method for being inline (also has loops inside)

File: Source/imgui.ini [hunk 5]
```diff
@@ -4,14 +4,13 @@ Size=47,53
 Collapsed=0
```
Comment (AlbertVVila) [inline Source/imgui.ini]: Not sure if you want this file to be modified

File: Source/AABBTree.cpp (hunk 3 above)
Comment (solidajenjo) [reply Source/AABBTree.cpp]: Solved

File: Source/AABBTree.h (hunk 4 above)
Comment (solidajenjo) [reply Source/AABBTree.h]: Solved

File: Source/imgui.ini (hunk 5 above)
Comment (solidajenjo) [reply Source/imgui.ini]: Reverted. Include in git ignore along config in develop if you can

**Comment by Rubetoman** [comment]
Ok

**Pull request merged**